pandas/core/index.py

```diff
@@ -2543,11 +2550,12 @@ def insert(self, loc, item):
         -------
         new_index : Index
         """
-        _self = np.asarray(self)
-        item_idx = Index([item], dtype=self.dtype).values
-        idx = np.concatenate(
-            (_self[:loc], item_idx, _self[loc:]))
-        return Index(idx, name=self.name)
+        indexes=[self[:loc],
+                 Index([item]),
+                 self[loc:]]
+
+        return indexes[0].append(indexes[1]).append(indexes[2])
```

Comment:
No, because you can insert a different type into the `Index` (e.g. put a string into an integer index), and `_shallow_copy` requires the same index type.
I struggled with this one for a while tbh - please suggest away if there's a better way of doing it.